### DISCUSSION TITLE
"No module named 'pyarabic'"

### DISCUSSION THREAD

**Discussion by Jammooly1** [deta/cloud-docs#654] [Help] [answered]
Whenever I run `deta run`

I always get this error:

```
Running micro...

Response:
{
        "errorMessage": "No module named 'pyarabic'",
        "errorType": "ModuleNotFoundError",
        "requestId": "971bb44c-1560-4db7-ad60-82f69843fcc6",
        "stackTrace": [
                "  File \"/opt/python/detalib/debugger.py\", line 131, in wrap\n    return func(event, context)\n",
                "  File \"/var/task/_entry.py\", line 12, in handler\n    import main  # noqa\n",
                "  File \"/var/task/main.py\", line 2, in \u003cmodule\u003e\n    from quran_sql import QuranDB\n",
                "  File \"/var/task/quran_sql.py\", line 3, in \u003cmodule\u003e\n    from pyarabic.araby import strip_tatweel, strip_tashkeel\n"
        ]
}

```
The original import statement looks like this `from pyarabic.araby import strip_tatweel, strip_tashkeel` but I already tried putting `pyarabic.araby` in requirements.txt but it still return the same error. I've modified it many other ways but to no avail. I have absolutely no idea how to tackle this. Deta docs needs an example of a requirements.txt with more than one library.

This is my requirments.txt currently:
```
fastapi
pyarabic
pydantic
sqlite3
```

So how do I fix this?

Thank you

**Comment by luisvivasg** [comment]
I tried and it works fine. It says that the module does not exist. Did you do the deta deploy? check the micro detail if the package exists and consequently installed.

**Reply by Jammooly1** [reply]
I ran `deta deploy` and I got this output which includes an error:

```
Updating dependencies...
Collecting pyarabic
  Downloading https://files.pythonhosted.org/packages/d7/64/0ea5be39e6a6515804cae8c280226d771f42750a08182f9d2e5f3b822694/PyArabic-0.6.15-py3-none-any.whl (126kB)
Collecting pydantic
  Downloading https://files.pythonhosted.org/packages/4c/5f/11db15638a3f5b29c7ae6f24b43c1e7985f09b0fe983621d7ef1ff722020/pydantic-1.10.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (13.2MB)
ERROR: Could not find a version that satisfies the requirement sqlite3 (from versions: none)
ERROR: No matching distribution found for sqlite3
You should consider upgrading via the 'pip install --upgrade pip' command.


Error: failed to update dependecies: error on one or more dependencies, no dependencies were added, see output for details
```

Also when I run `deta deploy` again, I'm getting:

```
Deploying...
Error: failed to deploy: Internal server error
```

It couldn't find the sqlite3 library even though I'm importing sqlite3 into my .py files like this `import sqlite3`. Do I only have to import the dependencies that are used with the keyword "from" such as `from pydantic import BaseModel`?

**Reply by Jammooly1** [reply]
I've tried and updated my requirements.txt in many different ways but to no avail. This is how it looks currently:

```
fastapi==0.88.0
PyArabic==0.6.15
pydantic==1.10.2
```

I ran `deta deploy purge-dependencies` then ran `deta deploy` but I still keep getting that same Internal server error without any details of informing what the issue is:

```
Deploying...
Error: failed to deploy: Internal server error
```

I

**Reply by Jammooly1** [reply]
```
deta new --project Texts_API
Error: failed to create new program: Internal server error
```

Are the servers healthy? I'm getting `Internal server error` for multiple commands.

**Comment by Jammooly1** [comment, marked as answer]
Managed to fix the issue. Part of it was due to their server being down. But in regards to requirements.txt, I had to get rid of sqlite3 as that is already apart of python and I didn't have to specify it. Then I ran deta deploy again and fortunately it was successful :)

**Comment by begoon** [comment]
Similar to https://github.com/deta/docs/discussions/658.

**Comment by serdarakyol** [comment]
I am having the same problem now. I have been waiting since morning but the problem didn't solve #658 

```bash
$ deta run
Running micro...

Response:
{
	"errorMessage": "No module named 'fastapi'",
	"errorType": "ModuleNotFoundError",
	"requestId": "21d9f63a-3997-4610-aaca-31c26fe77453",
	"stackTrace": [
		"  File \"/opt/python/detalib/debugger.py\", line 131, in wrap\n    return func(event, context)\n",
		"  File \"/var/task/_entry.py\", line 12, in handler\n    import main  # noqa\n",
		"  File \"/var/task/main.py\", line 1, in \u003cmodule\u003e\n    from fastapi import FastAPI\n"
	]
}
```
